### PR TITLE
Handle duplicates

### PR DESCRIPTION
**Pull request by berezkin88**

### CODE REVIEW

File: registry-manager-rest-server/src/main/java/de/adorsys/registry/manager/resource/AspspResource.java [hunk 1]
```diff
@@ -83,6 +83,16 @@ private AspspBO buildAspspBO(String name, String bic, String bankCode, String ad
                        .body(converter.toAspspTO(bo));
     }
 
+    @PreAuthorize("hasAnyRole('MANAGER','DEPLOYER')")
```
Comment (spetrychenko) [inline registry-manager-rest-server/src/main/java/de/adorsys/registry/manager/resource/AspspResource.java]: not sure this endpoint should be secured

File: registry-manager-service-api/src/main/java/de/adorsys/registry/manager/service/AspspService.java [hunk 2]
```diff
@@ -37,6 +37,14 @@
 
     AspspBO save(AspspBO aspsp);
 
+    /**
+     * Checks if an incoming Aspsp record is already represented in the database
+     *
+     * @param aspsp to check
+     * @return result if there are any duplicates
+     */
+    AspspBO checkNewAspsp(AspspBO aspsp);
```
Comment (spetrychenko) [inline registry-manager-service-api/src/main/java/de/adorsys/registry/manager/service/AspspService.java]: Seems to me this method has wrong signature. if the name started from "check" we are expecting boolean result but an object

File: registry-manager-service-impl/src/main/java/de/adorsys/registry/manager/service/impl/AspspServiceImpl.java [hunk 3]
```diff
@@ -105,4 +112,21 @@ public long count() {
 
         return repository.count();
     }
+
+    private AspspBO checkForDuplicates(AspspPO target) {
```
Comment (spetrychenko) [inline registry-manager-service-impl/src/main/java/de/adorsys/registry/manager/service/impl/AspspServiceImpl.java]: this impl proofs my previous comment

File: registry-manager-rest-server/src/main/java/de/adorsys/registry/manager/resource/AspspResource.java (hunk 1 above)
Comment (berezkin88) [reply registry-manager-rest-server/src/main/java/de/adorsys/registry/manager/resource/AspspResource.java]: Do you mean no security at all or a user can have any role?

File: registry-manager-service-api/src/main/java/de/adorsys/registry/manager/service/AspspService.java (hunk 2 above)
Comment (berezkin88) [reply registry-manager-service-api/src/main/java/de/adorsys/registry/manager/service/AspspService.java]: renamed

File: registry-manager-service-impl/src/main/java/de/adorsys/registry/manager/service/impl/AspspServiceImpl.java (hunk 3 above)
Comment (berezkin88) [reply registry-manager-service-impl/src/main/java/de/adorsys/registry/manager/service/impl/AspspServiceImpl.java]: renamed, thx

File: registry-manager-rest-server/src/main/java/de/adorsys/registry/manager/resource/AspspResource.java (hunk 1 above)
Comment (berezkin88) [reply registry-manager-rest-server/src/main/java/de/adorsys/registry/manager/resource/AspspResource.java]: done

File: registry-manager-rest-server/src/main/java/de/adorsys/registry/manager/model/AspspEquivalentsReportTO.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        return linesWithSimilarEntities.length == 0;
+    }
+
+    @Override
+    public String toString() {
+        return "AspspDuplicatsReportBO{" +
+            "aspsp=" + aspsp +
+            ", lines=" + Arrays.toString(linesWithSimilarEntities) +
+            '}';
+    }
```
Comment (berezkin88) [inline registry-manager-rest-server/src/main/java/de/adorsys/registry/manager/model/AspspEquivalentsReportTO.java:36]: lacking line

**Review by spetrychenko — APPROVED** [review]

**Pull request merged**